### PR TITLE
Compile Api and ProducerSideChannel on Windows

### PR DESCRIPTION
**Pull request by pierricgimmig**
Minor changes to enable the compilation of the API and
ProducerSideChannel projects on Windows. We now compile orbit.dll which
will be injected into a target process to enable manual instrumentation
which will first be used by the upcoming Windows API tracing feature.

Tests: compiles, runs in local Windows API tracing branch.

### CODE REVIEW

File: src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 namespace orbit_producer_side_channel {
 
+constexpr const char* kProducerSideWindowsServerAddress = "localhost:1789";
```
Comment (dpallotti) [inline src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h:16]: The constant that follows is a `std::string_view` and has a comment. Let's be consistent and let's have a comment for this too.

**Review by dpallotti — APPROVED** [review]
Looks good otherwise.

**Review by florian-kuebler — APPROVED** [review]
Very nice. Thanks!

File: src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h (hunk 1 above)
Comment (pierricgimmig) [reply src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h:16]: I'm not sure about using std::string_view for anything other than function parameters. We can't assign a std::string_view to a std::string for example, which I need here. I don't think there is an advantage to using `std::string_view` over `const char*` here, so I suggest replacing it in the original code.

File: src/Api/OrbitApiVersions.h [hunk 2]
```diff
@@ -45,6 +45,8 @@ struct orbit_api_v1 {
   void (*track_double)(const char* name, double value, orbit_api_color color);
 };
 
+#ifdef __linux
```
[the remark this answers is not in the document]
Comment (pierricgimmig) [reply src/Api/OrbitApiVersions.h:48]: Done, thanks.

File: src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h (hunk 1 above)
Comment (pierricgimmig) [reply src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h:18]: I replaced the original code, please let me know if I overlooked anything. Thanks.

Comment (dpallotti) [reply src/ProducerSideChannel/include/ProducerSideChannel/ProducerSideChannel.h:18]: I also prefer `const char*` on definition I guess, it's less surprising. But note:

> We can't assign a std::string_view to a std::string for example

You can with `std::string server_address{kProducerSideWindowsServerAddress}`. For reassignment `server_address = std::string{kProducerSideWindowsServerAddress}`. You only can't use `=` implicitly.

**Pull request merged**